crates/ln-dlc-node/src/seed.rs

```diff
@@ -8,17 +8,30 @@ use bip39::Language;
 use bip39::Mnemonic;
 use bitcoin::Network;
 use hkdf::Hkdf;
+use rand::thread_rng;
+use rand::RngCore;
 use sha2::Sha256;
 
+#[allow(unused)]
+const MIN_NB_WORDS: usize = 12;
+
+/// The maximum number of words in a mnemonic.
+const MAX_NB_WORDS: usize = 24;
+
 #[derive(Clone)]
 pub struct Bip39Seed {
     mnemonic: Mnemonic,
 }
 
 impl Bip39Seed {
     pub fn new() -> Result<Self> {
-        let mut rng = rand::thread_rng();
-        let mnemonic = Mnemonic::generate_in_with(&mut rng, Language::English, 12)?;
+        let word_count = 12;
+
+        let entropy_bytes = (word_count / 3) * 4;
+        let mut entropy = [0u8; (MAX_NB_WORDS / 3) * 4];
+        thread_rng().fill_bytes(&mut entropy[0..entropy_bytes]);
+
+        let mnemonic = Mnemonic::from_entropy_in(Language::English, &entropy[0..entropy_bytes])?;
```

Comment:
Can you explain this please?

Reply:
We copied this peace of code over from bip39 as the bip39 dependency uses an outdated rand lib. This was the fastest way to bypass the conflicting rand dependency. @luckysori wanted to look into creating our own fork of bip39 fixing that dependency issue the right way.

Reply:
I've updated this now so that we don't have to inline this code.

Reply:
Thanks for the explanation